### PR TITLE
(ci): Convert OBP Signed Binaries Pipeline to ADOv2 Yaml

### PR DESCRIPTION
**Pull request by sheylatrudo**
**Reason for Change**:
The OBP Signed Binaries pipeline is currently a v1 ADO Release pipeline. We have some upcoming requirements to migrate all v1 type pipelines to v2 - yaml style - pipelines. 

**Issue Fixed**:

Completes the majority of the work for [removed]

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by estebancams — APPROVED** [review]

**Review by chengch801214 — APPROVED** [review]

File: .pipelines/obp-signed-binaries-pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      branches:
+        include:
+          - master
+  repositories:
+  # TODO: Change to an official service connection for integration with other pipelines.
+  - repository: acn_master
+    endpoint: "sheylatrudo"
+    type: github
+    name: sheylatrudo/azure-container-networking
+    ref: smtrudo/obp-signed-binaries-pipeline
```
Comment (rbtr) [inline .pipelines/obp-signed-binaries-pipeline.yaml:47]: I don't think this pipeline ref to a fork is correct

File: .pipelines/obp-signed-binaries/publish-nuget-packages.stages.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    VERSION: $[ stageDependencies.prepare_stage.prepare_version.outputs['get_version.version'] ]
+  jobs:
+  - job: 
+    displayName: Publish Nuget Packages
+    workspace:
+      clean: all
+    strategy:
+      matrix:
+        azure_cni:
+          nuget_feed: "b32aa71e-8ed2-41b2-9d77-5bc261222004/95d6b69b-00c0-4d30-98f9-122e08393820"
```
Comment (rbtr) [inline .pipelines/obp-signed-binaries/publish-nuget-packages.stages.yaml:31]: are these feeds public?

File: .pipelines/obp-signed-binaries/publish-nuget-packages.stages.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    - checkout: none
+
+    - download: networking_aquarius_official
+      artifact: $(artifact)
+      patterns: $(file_pattern)
+
+    - task: NuGetAuthenticate@1
+      displayName: Authenticate to Nuget
+
+    # A test step. Can be removed once integrated with the regular pipeline.
```
Comment (rbtr) [inline .pipelines/obp-signed-binaries/publish-nuget-packages.stages.yaml:75]: should be removed now?

File: .pipelines/obp-signed-binaries-pipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  - none
+
+resources:
+  pipelines:
+  - pipeline: networking_aquarius_official
+    source: "Networking-Aquarius-Official"
+    trigger:
+      branches:
+        include:
+          - master
```
Comment (jpayne3506) [inline .pipelines/obp-signed-binaries-pipeline.yaml:40]: Do we want this pipeline to trigger on every merge to master?

**Review by rbtr — CHANGES_REQUESTED** [review]

**Comment by rbtr** [comment]
TBH I don't know what this is or why we need it, and the PR description seems to assume I know that already. But even just from reviewing it I'm skeptical that it belongs in ACN rather than NA, given the apparently sensitive and non-public nature of some of the actions here. 
This repo is _public_. Internal references, even as innocuous as linking our ADO items, is not appropriate. So pipelines publishing private nuget feeds definitely shouldn't be here.

File: .pipelines/obp-signed-binaries-pipeline.yaml (hunk 1 above)
Comment (sheylatrudo) [reply .pipelines/obp-signed-binaries-pipeline.yaml:47]: Not a functional pipeline, it's for reference. Someone else to complete the work TBD.

File: .pipelines/obp-signed-binaries/publish-nuget-packages.stages.yaml (hunk 3 above)
Comment (sheylatrudo) [reply .pipelines/obp-signed-binaries/publish-nuget-packages.stages.yaml:75]: No. this is not a functional pipeline and was being added so the work could be picked up.

**Pull request closed without merging**